### ISSUE TITLE
wmlparser3 should not emit invalid WML

### ISSUE THREAD

**Issue by soliton-** [wesnoth/wesnoth#7653]
When calling `wmlparser3.py -t '# foo'` the produced output is `# foo=` which is not valid WML. This should produce a WMLError instead.

From a quick test it looks like several other inputs (`]`, `foo bar`) also cause invalid output instead of an error.

**Comment by soliton-** [comment]
It's not. Feel free to try it (obviously without preprocessing) or perhaps read https://wiki.wesnoth.org/GrammarWML#WML.

**Comment by ProditorMagnus** [comment]
Confirmed, and also discovered unrelated 1.17 bug.

**Comment by CelticMinstrel** [comment]
> I think `# foo=` is valid WML, just not the expected result from parsing the source.

It's a valid input to the WML preprocessor, but it's possibly not a valid input to the WML parser? The preprocessor is supposed to strip out comments, so the parser doesn't have to deal with them.

**Comment by ProditorMagnus** [comment]
Yes, I just never noticed lack of comments after generating preprocessed addon.
